llvm/test/CodeGen/AArch64/setcc_knownbits.ll

```diff
@@ -21,9 +21,7 @@ define noundef i1 @logger(i32 noundef %logLevel, ptr %ea, ptr %pll) {
 ; CHECK-NEXT:    ret
 ; CHECK-NEXT:  .LBB1_2: // %land.rhs
 ; CHECK-NEXT:    ldr x8, [x1]
-; CHECK-NEXT:    ldrb w8, [x8]
-; CHECK-NEXT:    cmp w8, #0
-; CHECK-NEXT:    cset w0, ne
+; CHECK-NEXT:    ldrb w0, [x8]
```

Comment:
Not enough test updates? I would expect a diff from the zext/sext/anyext cases

Reply:
Perhaps it's possible to use the range metadata to write some explicit tests for sext/anyext?

Reply:
Yeah I need to add some better tests. Whilst this seems to come up a fair amount during C++ code (bools often have range metadata applied), it can be difficult to test is simple cases as the `sext(load !range)` is often already optimized before it creates the `load sext !range`. It might require C++ testing.